pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -152,7 +154,7 @@ func (s *Server) initDNSCerts(hostname string) error {
 
 	// Save the certificates to ./var/run/secrets/istio-dns - this is needed since most of the code we currently
 	// use to start grpc and webhooks is based on files. This is a memory-mounted dir.
-	if err := os.MkdirAll(dnsCertDir, 0600); err != nil {
+	if err := os.MkdirAll(dnsCertDir, 0700); err != nil {
```

Comment:
nit: do we need `+x`?

Reply:
The subsequent `ioutil.WriteFile` were failing with a permission denied. `+x` is needed to change to the directory and access files within it. 

```bash
$ mkdir test-dir
$ chmod 0600 test-dir/ 
$ touch test-dir/a
touch: cannot touch 'test-dir/a': Permission denied
$ chmod 07000 test-dir/
$ touch test-dir/a
$ echo $?
0
```